Add ability to set combat requirements for achievement diaries

Monsterxsync opened the pull request:
Added ability to add combat requirements to diarys also added all the current combat requirements.

ScreenShots:
When you meet the combat requirement.
![when you have requirement](https://puu.sh/BTHtl/35755b3003.png)
When you don't meet the combat requirement.
![when you don't have requirement](https://puu.sh/BTHtM/ce5c50ef33.png)

Nightfirecat commented on runelite-api/src/main/java/net/runelite/api/Skill.java:
```diff
@@ -56,7 +56,8 @@
 	 */
-	OVERALL("Overall");
+	OVERALL("Overall"),
+	COMBAT("Combat");
```
This is a very poor way to implement this feature. This will cause very bad side effects for any plugin which tries to iterate over `Skill.values()` such as the XP tracker plugin. This should instead be incorporated into `Requirement` somehow.

Monsterxsync replied:
wouldn't overall cause issues with that too?, i checked if it interfered with the xp tracker and didnt see an issue.

Monsterxsync commented:
I've moved it out of the `Skill Enum` and added a 2nd constructor to `Requirement`, it just assumes that you're using the custom field for combat requirement if i can think of another use for the custom reqs i'll change it around.

Nightfirecat commented on runelite-client/src/main/java/net/runelite/client/plugins/achievementdiary/DiaryRequirementsPlugin.java:
```diff
@@ -242,7 +244,25 @@ private GenericDiaryRequirement getRequirementsForTitle(String title)
+				Skill skill = i.getSkill();
+				int realSkillLevel;
+				//if skill is null assumed to be combat.
+				if (skill == null)
```
This should probably also check that the custom requirement is "Combat".

Nightfirecat commented on runelite-client/src/main/java/net/runelite/client/plugins/achievementdiary/DiaryRequirementsPlugin.java:
```diff
@@ -242,7 +244,25 @@ private GenericDiaryRequirement getRequirementsForTitle(String title)
+				//if skill is null assumed to be combat.
+				if (skill == null)
+				{
+					realSkillLevel =  Experience.getCombatLevel(
```
`realSkillLevel = client.getLocalPlayer().getCombatLevel();`

Monsterxsync replied:
lol, i knew there had to be an easier way to grab the players combat level.

Nightfirecat commented on runelite-client/src/main/java/net/runelite/client/plugins/achievementdiary/Requirement.java:
```diff
@@ -29,14 +29,22 @@
-	private final Skill skill;
-	private final int levelRequirement;
-	private final Requirement[] altRequirements;
+	private Skill skill;
```
Make these final and `null` the requirement type which isn't set. (skill or customRequirement, that is)
